abstractprotocol/protocol.go

```diff
@@ -575,7 +575,7 @@ func addNodeBuiltInProps(pol *policy.Policy) (*policy.Policy, error) {
 	// get built-in node properties and replace the ones in the policy,
 	// the memory will be the available memory instead of total memory -- not yet,
 	//still use total memory, change to true if you want available memory
-	builtinNodePol := externalpolicy.CreateNodeBuiltInPolicy(false)
+	builtinNodePol := externalpolicy.CreateNodeBuiltInPolicy(false, false)
```

Comment:
I think this is a problem, if Create... generates a new hardwareId every time, then the hardwareId couldnt be used in a business policy constraint. I think we need the hardwareId value from the node itself at this point.